[immer] CMake `find_package(immer VERSION 0.8.0)` fails because of missing versioning info

**Is your feature request related to a problem? Please describe.**
In CMake, running `find_package(immer "0.8.0" REQUIRED)` fails because the version info is unset.

```log
-- Running vcpkg install - done
CMake Error at C:/vcpkg/scripts/buildsystems/vcpkg.cmake:844 (_find_package):
  Could not find a configuration file for package "immer" that is compatible
  with requested version "0.8.0".

  The following configuration files were considered but not accepted:

    C:/Users/mypc/Projects/hello/cmake-build-debug/vcpkg_installed/x64-windows/share/immer/immerConfig.cmake, version: unknown

Call Stack (most recent call first):
  CMakeLists.txt:109 (find_package)


ninja: error: rebuilding 'build.ninja': subcommand failed
-- Configuring incomplete, errors occurred!
See also "C:/Users/mypc/Projects/hello/cmake-build-debug/CMakeFiles/CMakeOutput.log".
```

**Proposed solution**
Add versioning info to `immer`'s CMake module. (Unlike other packages, it seems to be missing `vcpkg_installed\x64-windows\share\immer\ImmerConfigVersion.cmake`.)

--- a/ports/immer/fix-ExportConfigVersion.patch
+++ b/ports/immer/fix-ExportConfigVersion.patch
@@ -1,0 +1,19 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a8cfbd3..84c89a0 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -96,6 +96,14 @@ target_include_directories(immer INTERFACE
+   $<BUILD_INTERFACE:${immer_SOURCE_DIR}/>
+   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+ install(TARGETS immer EXPORT ImmerConfig)
++
++include(CMakePackageConfigHelpers)
++write_basic_package_version_file(
++  ${CMAKE_CURRENT_BINARY_DIR}/ImmerConfigVersion.cmake
++  VERSION ${PROJECT_VERSION}
++  COMPATIBILITY SameMajorVersion )
++
++install(FILES "${CMAKE_CURRENT_BINARY_DIR}/ImmerConfigVersion.cmake" DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Immer" )
+ install(EXPORT ImmerConfig DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Immer")
+ install(DIRECTORY immer DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+ 

--- a/ports/immer/fix-ExportConfigVersion.patch
+++ b/ports/immer/fix-ExportConfigVersion.patch
@@ -9,7 +9,7 @@ index a8cfbd3..84c89a0 100644
 +
 +include(CMakePackageConfigHelpers)
 +write_basic_package_version_file(
-+  ${CMAKE_CURRENT_BINARY_DIR}/ImmerConfigVersion.cmake
++  "${CMAKE_CURRENT_BINARY_DIR}/ImmerConfigVersion.cmake"
 +  VERSION ${PROJECT_VERSION}
 +  COMPATIBILITY SameMajorVersion )
 +

--- a/ports/immer/portfile.cmake
+++ b/ports/immer/portfile.cmake
@@ -4,9 +4,10 @@ vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO arximboldi/immer
-    REF v${VERSION}
+    REF "v${VERSION}"
     SHA512 fc34242d36efdb9faa1f22ccc7591c1ace34c2b383e1266a290346baedc154e3d4a682d6dd5094460b75e123347194710072e996d19165cc5fd23c922fdfc4e8
     HEAD_REF master
+    PATCHES fix-ExportConfigVersion.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/immer/portfile.cmake
+++ b/ports/immer/portfile.cmake
@@ -1,5 +1,4 @@
 # header-only library
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
@@ -7,7 +6,8 @@ vcpkg_from_github(
     REF "v${VERSION}"
     SHA512 fc34242d36efdb9faa1f22ccc7591c1ace34c2b383e1266a290346baedc154e3d4a682d6dd5094460b75e123347194710072e996d19165cc5fd23c922fdfc4e8
     HEAD_REF master
-    PATCHES fix-ExportConfigVersion.patch
+    PATCHES
+        fix-ExportConfigVersion.patch # Upstream PR https://github.com/arximboldi/immer/pull/250 has been merged, this patch need to be removed in next update.
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/immer/vcpkg.json
+++ b/ports/immer/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "immer",
   "version": "0.8.0",
+  "port-version": 1,
   "description": "Postmodern immutable and persistent data structures for C++",
   "homepage": "https://sinusoid.es/immer/",
   "license": "BSL-1.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3130,7 +3130,7 @@
     },
     "immer": {
       "baseline": "0.8.0",
-      "port-version": 0
+      "port-version": 1
     },
     "implot": {
       "baseline": "0.14",

--- a/versions/i-/immer.json
+++ b/versions/i-/immer.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c5e5b8f96f6f913707c803a2adb02903e986338a",
+      "version": "0.8.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "9e996304abce149829a2d0db1156bcf34c9e51c8",
       "version": "0.8.0",
       "port-version": 0

--- a/versions/i-/immer.json
+++ b/versions/i-/immer.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c5e5b8f96f6f913707c803a2adb02903e986338a",
+      "git-tree": "ce97c8a31a8c00fe6a8ee5d8a14db013134b6ce4",
       "version": "0.8.0",
       "port-version": 1
     },

--- a/versions/i-/immer.json
+++ b/versions/i-/immer.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ce97c8a31a8c00fe6a8ee5d8a14db013134b6ce4",
+      "git-tree": "e133c376f7ef7d4e7c8b78245b70231e8df5b601",
       "version": "0.8.0",
       "port-version": 1
     },

